### PR TITLE
NavigationRail

### PR DESCRIPTION
**Pull request by alsturm**
grafik

### CODE REVIEW

File: src/modules/chips/components/chipsContainer/ChipsContainer.js [hunk 1]
```diff
@@ -177,11 +185,13 @@ export class ChipsContainer extends MvuElement {
...
 		};
 
+		const getOverlayNavClass = () => (isOpenNavigationRail && !isPortrait ? 'is-open-navigationRail' : '');
+
 		return html`
 			<style>
 				${css}
 			</style>
-			<div id="chipscontainer" class="${getOrientationClass()} ${getMinWidthClass()} ${getOverlayClass()} chips__container">
+			<div id="chipscontainer" class="${getOrientationClass()} ${getMinWidthClass()} ${getOverlayClass()} ${getOverlayNavClass()} chips__container">
```
Comment (thiloSchlemmer) [inline src/modules/chips/components/chipsContainer/ChipsContainer.js:194]: I suggest to collect all classes in a classMap, to have the controlflow in one place like:
```suggestion
    const classes = {
            'is-open-navigationRail':isOpenNavigationRail && !isPortrait,
            'is-desktop': hasMinWidth,
            'is-tablet':!hasMinWidth,
            'is-portrait' : isPortrait,
            'is-landscape':!isPortrait
```
}`

File: src/modules/footer/components/Footer.js [hunk 2]
```diff
@@ -94,7 +104,7 @@ export class Footer extends MvuElement {
 			</style>
 			<div class="preload">
 				<div class="${getOrientationClass()} ${getMinWidthClass()} ${isEmbedded()}">
```
Comment (thiloSchlemmer) [inline src/modules/footer/components/Footer.js:106]: same suggest as in ChipsContainer.js --> use of ${classMap(classes)}

File: src/modules/header/components/Header.js [hunk 3]
```diff
@@ -231,9 +244,9 @@ export class Header extends MvuElement {
 		return html`
 			<style>${css}</style>
 			<div class="preload">
-				<div class="${getOrientationClass()} ${getMinWidthClass()} ${getDemoClass()}">
+				<div class="${getOrientationClass()} ${getMinWidthClass()} ${getDemoClass()} ${getOverlayTestClass()}">
```
Comment (thiloSchlemmer) [inline src/modules/header/components/Header.js:247]: same suggest as in ChipsContainer.js --> use of ${classMap(classes)}

File: src/modules/menu/components/mainMenu/MainMenu.js [hunk 4]
```diff
@@ -169,7 +179,7 @@ export class MainMenu extends MvuElement {
 				${css}
 			</style>
 			<div class="${getOrientationClass()} ${getPreloadClass()}">
-				<div id="mainmenu" class="main-menu ${getOverlayClass()} ${getMinWidthClass()} ${getFullSizeClass()}">
+				<div id="mainmenu" class="main-menu ${getOverlayClass()} ${getOverlayNavClass()} ${getMinWidthClass()} ${getFullSizeClass()}">
```
Comment (thiloSchlemmer) [inline src/modules/menu/components/mainMenu/MainMenu.js:182]: same suggest as in ChipsContainer.js --> use of ${classMap(classes)}

File: src/modules/menu/components/navigationRail/NavigationRail.js [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+			//first tow are always home-button and separator
+			return reverseTabIds.indexOf(tabId) + 2;
+		};
+
+		const translate = (key) => this._translationService.translate(key);
+		return html`
+			<style>
+				${css}
+			</style>
+			<div class="navigation-rail__container ${getOverlayClass()} ${getOrientationClass()}">
```
Comment (thiloSchlemmer) [inline src/modules/menu/components/navigationRail/NavigationRail.js:144]: same suggest as in ChipsContainer.js --> use of ${classMap(classes)}

File: src/modules/stackables/components/bottomSheet/BottomSheet.js [hunk 6]
```diff
@@ -72,10 +82,14 @@ export class BottomSheet extends MvuElement {
...
+			return portrait ? 'is-portrait' : 'is-landscape';
+		};
+
 		return content
 			? html`
 		<style>${css}</style>
-		<div class='bottom-sheet ${getOverlayClass()}' data-test-id>
+		<div class='bottom-sheet ${getOverlayClass()} ${getOverlayNavClass()} ${getOrientationClass()}' data-test-id>
         	${content}
 			<ba-icon id="close-icon" class='tool-container__close-button' .icon='${closeIcon}' .size=${1.6} .color=${'var(--text2)'} .color_hover=${'var(--text2)'} @click=${onDismiss}>
```
Comment (thiloSchlemmer) [inline src/modules/stackables/components/bottomSheet/BottomSheet.js:94]: Unclosed tag

File: src/modules/stackables/components/bottomSheet/BottomSheet.js [hunk 7]
```diff
@@ -72,10 +82,14 @@ export class BottomSheet extends MvuElement {
...
 
+		const getOrientationClass = () => {
+			return portrait ? 'is-portrait' : 'is-landscape';
+		};
+
 		return content
 			? html`
 		<style>${css}</style>
-		<div class='bottom-sheet ${getOverlayClass()}' data-test-id>
+		<div class='bottom-sheet ${getOverlayClass()} ${getOverlayNavClass()} ${getOrientationClass()}' data-test-id>
```
Comment (thiloSchlemmer) [inline src/modules/stackables/components/bottomSheet/BottomSheet.js:92]: same suggest as in ChipsContainer.js --> use of ${classMap(classes)}

File: src/plugins/NavigationRailPlugin.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+/**
+ * @module plugins/NavigationRailPlugin
+ */
+import { observe } from '../utils/storeUtils';
+import { BaPlugin } from '../plugins/BaPlugin';
+import { open, addTabId } from '../store/navigationRail/navigationRail.action';
+import { TabIds } from '../domain/mainMenu';
+
+/**
+ * @class
```
Comment (thiloSchlemmer) [inline src/plugins/NavigationRailPlugin.js:10]: Description missing

File: test/e2e/page.spec.js [hunk 9]
```diff
@@ -63,7 +63,7 @@ test.describe('page', () => {
 		});
 
 		test('should contain 14 top level ba-components', async ({ page }) => {
-			expect(await page.locator('body > *').count()).toBe(14);
+			expect(await page.locator('body > *').count()).toBe(15);
```
Comment (thiloSchlemmer) [inline test/e2e/page.spec.js:66]: a specific test for the component is missing.

```suggestion
			expect(await page.locator('ba-navigation-rail').count()).toBe(1);
```

File: src/modules/menu/i18n/menu.provider.js [hunk 10]
```diff
@@ -51,7 +60,16 @@ export const provide = (lang) => {
 				menu_misc_content_panel_ea_header: 'Energie-Atlas Bayern',
 				menu_misc_content_panel_ea_text: 'Das zentrale Internet-Portal zum Energiesparen, zur Energieeffizienz und zu erneuerbaren Energien.',
 				menu_misc_content_panel_feedback_title: 'Feedback',
-				menu_misc_content_panel_routing_title: 'Routing'
+				menu_misc_content_panel_routing_title: 'Routing',
+
+				menu_navigation_rail_home: 'Home',
+				menu_navigation_rail_routing: 'Routine',
+				menu_navigation_rail_close: 'schließen',
+				menu_navigation_rail_zoom_to_extend: 'auf Bayern Zoomen',
```
Comment (thiloSchlemmer) [inline src/modules/menu/i18n/menu.provider.js:68]: auf Bayern (z)oomen

File: test/service/StoreService.test.js [hunk 11]
```diff
@@ -127,7 +131,7 @@ describe('StoreService', () => {
 			expect(store).toBeDefined();
 
 			const reducerKeys = Object.keys(store.getState());
-			expect(reducerKeys.length).toBe(28);
+			expect(reducerKeys.length).toBe(29);
```
Comment (thiloSchlemmer) [inline test/service/StoreService.test.js:134]: There  should be a specific test for the 'navigationRail' reducer, too.

```suggestion
			expect(reducerKeys.includes('navigationRail')).toBeTrue();
```

**Review by thiloSchlemmer — CHANGES_REQUESTED** [review]
Nice component!
Some minor suggestions.

File: src/modules/menu/components/navigationRail/NavigationRail.js [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+import { close } from '../../../../store/navigationRail/navigationRail.action';
+
+const Update_IsOpen_TabIndex = 'update_isOpen_tabIndex';
+const Update_IsOpen_NavigationRail = 'update_NavigationRail';
+const Update_IsPortrait_HasMinWidth = 'update_isPortrait_hasMinWidth';
+const Update_Schema = 'update_schema';
+const Update_FeatureInfo_Data = 'update_featureInfo_data';
+
+/**
+ *
```
[the remark this answers is not in the document]
Comment (taulinger) [reply src/modules/menu/components/navigationRail/NavigationRail.js:24]: Have a look at our technical docs:
grafik

**Review by thiloSchlemmer — APPROVED** [review]
Thx

**Pull request merged**